### PR TITLE
feat: add a flag to bypass profile cache

### PR DESCRIPTION
**Pull request by juanmahidalgo**
#313

### CODE REVIEW

File: src/lib/types.ts [hunk 1]
```diff
@@ -38,3 +38,7 @@ export type StorageOwnData = {
     version: number
   }
 }
+
+export type FetchProfileOptions = {
+  useCache: boolean
```
Comment (LautaroPetaccio) [inline src/lib/types.ts:43]: As it is a set of options, should we mark them as optional as well?
```suggestion
  useCache?: boolean
```

File: src/lib/peer.spec.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          } as Profile
+          peerApi.cache = {}
+          jest
+            .spyOn(peerApi.lambdasClient, 'fetchProfiles')
+            .mockImplementation(() => Promise.resolve([profileUpdated]))
+        })
+        it('should fetch the profile', async () => {
+          expect(await peerApi.fetchProfile(address, { useCache: true })).toBe(
+            profileUpdated
+          )
```
Comment (LautaroPetaccio) [inline src/lib/peer.spec.ts:59]: Is it possible to check that the cache was updated as well? Like, for example, we might be able to perform an:
```
await expect(peerApi.cache).resolves.toEqual(profileUpdated)
```
What if we add something like this to ensure the cache to be set?
The same applies for the case when the `useCache` parameter is not passed.

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]
Great job 👍

File: src/lib/types.ts (hunk 1 above)
Comment (juanmahidalgo) [reply src/lib/types.ts]: updated!

File: src/lib/peer.spec.ts (hunk 2 above)
Comment (juanmahidalgo) [reply src/lib/peer.spec.ts:59]: added! in the case the `useCache` parameter is not used, the cache is set anyways, so the `expect` is the same in both cases.

**Review by LautaroPetaccio — APPROVED** [review]

**Comment by LautaroPetaccio** [comment]
Goes out with 🍟 , well done!

**Pull request merged**